component/componenttest/nop_exporter.go

```diff
@@ -35,54 +34,31 @@ type nopExporterConfig struct {
 	config.ExporterSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct
 }
 
-// nopExporterFactory is factory for nopExporter.
-type nopExporterFactory struct {
-	internalinterface.BaseInternal
-}
-
-var nopExporterFactoryInstance = &nopExporterFactory{}
+var nopExporterFactory = component.NewExporterFactory(
+	"nop",
+	func() config.Exporter {
+		return &nopExporterConfig{
+			ExporterSettings: config.NewExporterSettings(config.NewComponentID("nop")),
+		}
+	},
+	component.WithTracesExporter(createTracesExporter),
+	component.WithMetricsExporter(createMetricsExporter),
+	component.WithLogsExporter(createLogsExporter))
 
 // NewNopExporterFactory returns a component.ExporterFactory that constructs nop exporters.
 func NewNopExporterFactory() component.ExporterFactory {
```

Comment:
I believe the function should no longer return a global instance, but I would like to look at this separately, since this PR does not change anything. I will file an issue tomorrow.

Reply:
https://github.com/open-telemetry/opentelemetry-collector/issues/4906